crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3981,6 +3985,28 @@ mod tests {
         assert_public_ty(&db, "src/a.py", "d", "Literal[-1]");
         assert_public_ty(&db, "src/a.py", "e", "Literal[2]");
 
+        db.write_dedented(
+            "/src/b.py",
+            &format!(
+                "
+                a = 1 / 2
+                b = 1 // 2
+                c = 4 / 2
+                d = 4 // 2
+                e = 4 // 0    # division by zero
+                f = {} // -1  # overflow
+                ",
+                i64::MIN
+            ),
+        )?;
+
+        assert_public_ty(&db, "/src/b.py", "a", "float");
+        assert_public_ty(&db, "/src/b.py", "b", "Literal[0]");
+        assert_public_ty(&db, "/src/b.py", "c", "float");
+        assert_public_ty(&db, "/src/b.py", "d", "Literal[2]");
+        assert_public_ty(&db, "/src/b.py", "e", "int");
+        // TODO: Determine why the division overflow isn't an int
+        assert_public_ty(&db, "/src/b.py", "f", "Unknown");
```

Comment:
You're hitting exactly the overflow you expect, but you're hitting it sooner than you expect, because `i64::MIN` is a negative number, meaning in the Python AST it turns into a positive number with unary minus, and the positive number overflows i64, so it falls back to `int` instead of `LiteralInt` already in `infer_number_literal_expression`.

But then our current implementation of `infer_unary_expression` only has cases for literals, and otherwise falls back to `Unknown`, so by the time you get to your floor division, one of the operands is already unknown.

So I don't think you can use this case to check division overflow; it's really a test of integer-literal overflow and unary-op handling of non-literal types. I don't think there is actually any way to hit division-overflow with literal integer operands, so I would just remove this from the test.